### DISCUSSION TITLE
How to override styles generated by VE using css

### DISCUSSION THREAD

**Discussion by Yaojian** [vanilla-extract-css/vanilla-extract#1257] [Q&A] [answered]
I am developing an application which allows users to load CSS fragment dynamically to customize the UI.

As the class name is generated dynamically and may be changed over time, is it possible to keep some of the class name unchanged, so user can override them using the traditional CSS techniques?

**Comment by askoufis** [comment, marked as answer]
If it's just a few styles, then you could use [`globalStyle`](https://vanilla-extract.style/documentation/global-api/global-style/) to create a global classname, e.g.:

```ts
globalStyle('.stableClassname', {
  color: 'red'
});
```

However this comes with some restrictions compared to the `style` API.

Alternatively you could generate [custom identifiers](https://vanilla-extract.style/documentation/integrations/vite/#identifiers) that don't have a `hash` in them, or alternatively provide your own hash that's still unique, but more stable.
